### ISSUE TITLE
`no-extra-parens` should complain about unnecessary parenthesis for arrow functions.

### ISSUE THREAD

**Issue by TheGrandmother** [eslint/eslint#8736]
The `no-extra-parens` rule does as of now not complain about unnecessary parentheses for unary arrow functions.

Eslint will consider this code to be fine:
```
var foo = (x) => x + 1
```
even though the parentheses around the arguments are unnecessary and
```
var foo = x => x + 1
```
works just fine.

I suggest improving the rule or adding an option to check for this.

**Env**
Node: v8.1.0
npm: 5.0.3
eslint: 3.19.0

**Comment by not-an-aardvark** [comment]
Thanks for the report. This is working as intended -- you can use the [`arrow-parens`](http://eslint.org/docs/rules/arrow-parens) rule to check for this.

**Comment by TheGrandmother** [comment]
Oh! Thanks!
Sorry for prematurely creating an issue.

**Comment by platinumazure** [comment]
I think we do have a problem here-- the rule docs don't mention that arrow function parameter parens are always ignored (regex literals and IIFEs are explicitly listed as always ignored). In addition, the Related Rules section does not mention the `arrow-parens` rule.

I think we could accept a documentation contribution to fix those issues. I would like to reopen and change this to a docs issue. @not-an-aardvark Thoughts?

**Comment by not-an-aardvark** [comment]
Sounds good to me.

**Comment by platinumazure** [comment]
Nah, on second though I'll just create a new issue to avoid confusion.

Sorry for the noise.
